utils/stall_free.hh

```diff
@@ -57,14 +59,176 @@ void merge_to_gently(std::list<T>& list1, const std::list<T>& list2, Compare com
     }
 }
 
-template<class T>
-seastar::future<> clear_gently(std::list<T>& list) {
-    return repeat([&list] () mutable {
-        if (list.empty()) {
-            return seastar::stop_iteration::yes;
-        }
-        list.pop_back();
-        return seastar::stop_iteration::no;
+// The clear_gently functions are meant for
+// gently destroying the contents of containers.
+// The containers can be re-used after clear_gently
+// or may be destroyed.  But unlike e.g. std::vector::clear(),
```

Comment:
It took me a while to understand the sentence "the container can be re-used or may be destroyed" - it's sounds like this covers all the options :-) Eventually I realized  you just meant to say that clear_gently() leaves the container empty - and the user may with this empty container whatever he/she wants - this obviously means the user can relill it with data or destroy it.

Reply:
right.